File: mixin-base/src/main/java/net/yakclient/mixin/base/internal/methodadapter/core/CoreBIPatternMatcher.java
package net.yakclient.mixin.base.internal.methodadapter.core;

import net.yakclient.mixin.base.internal.instruction.Instruction;
import org.objectweb.asm.Handle;
import org.objectweb.asm.MethodVisitor;

public class CoreBIPatternMatcher extends CoreMixinPatternMatcher {
    public CoreBIPatternMatcher(MethodVisitor visitor, Instruction instructions) {
        super(visitor, instructions);
    }

    @Override
    public void visitMethodInsn(int opcode, String owner, String name, String desc) {
        this.executeInsn();
        super.visitMethodInsn(opcode, owner, name, desc);
    }

    @Override
    public void visitMethodInsn(int opcode, String owner, String name, String desc, boolean itf) {
        this.executeInsn();
        super.visitMethodInsn(opcode, owner, name, desc, itf);
    }

    @Override
    public void visitInvokeDynamicInsn(String name, String desc, Handle bsm, Object... bsmArgs) {
        this.executeInsn();
        super.visitInvokeDynamicInsn(name, desc, bsm, bsmArgs);
    }

    @Override
    public void visitInsn() {
    }
}

File: mixin-base/src/main/kotlin/net/yakclient/mixin/base/internal/methodadapter/core/CoreBIPatternMatcher.kt
package net.yakclient.mixin.base.internal.methodadapter.core

import net.yakclient.mixin.base.internal.instruction.Instruction
import org.objectweb.asm.Handle
import org.objectweb.asm.MethodVisitor

class CoreBIPatternMatcher(
    visitor: MethodVisitor, instructions: Instruction
) : CoreMixinPatternMatcher(visitor, instructions) {

    override fun visitMethodInsn(opcode: Int, owner: String, name: String, desc: String) {
        this.executeInsn()
        super.visitMethodInsn(opcode, owner, name, desc)
    }

    override fun visitMethodInsn(opcode: Int, owner: String, name: String, desc: String, itf: Boolean) {
        this.executeInsn()
        super.visitMethodInsn(opcode, owner, name, desc, itf)
    }

    override fun visitInvokeDynamicInsn(name: String, desc: String, bsm: Handle, vararg bsmArgs: Any) {
        this.executeInsn()
        super.visitInvokeDynamicInsn(name, desc, bsm, *bsmArgs)
    }

    override fun visitInsn() {}
}